### PR TITLE
review apps

### PR DESCRIPTION
**Pull request by zimbatm**

### CODE REVIEW

**Comment by zimbatm** [comment]
> Skipping a deployment with the script provider because the current build is a pull request.

hmm no, that's not what I want Mr Travis

**Comment by zimbatm** [comment]
It looks like the Travis Deployment context doesn't have nix installed...

**Comment by zimbatm** [comment]
alright, this is ready

File: swarm/deploy [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash
+# vim: ft=sh ts=2 sw=2 et
+#
+# Usage: deploy [PULL_REQUEST]
```
Comment (nmattia) [inline swarm/deploy]: what's PULL_REQUEST? the branch name, the PR number?

File: swarm/lib.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#!bash
+#
+# A collection of utilities for the review and deployments
+set -euo pipefail
```
Comment (nmattia) [inline swarm/lib.sh]: if this is sourced from another script, `set`ing might mess stuff up

File: travis/deploy [hunk 3]
```diff
@@ -2,40 +2,31 @@
...
+# See https://github.com/travis-ci/travis-ci/issues/7552
+export TRAVIS_BUILD_URL=https://travis-ci.com/$TRAVIS_REPO_SLUG/builds/$TRAVIS_BUILD_ID
 
 set -x
-curl -u "$SWARM_INTELLIGENCE_AUTH" -F name=remote-tabs -F "file=@$tmpfile" https://intelligence.netice9.com/api/deploy_stack
+github-deploy please \
+  --deploy-script swarm/deploy \
+  --build-url "$TRAVIS_BUILD_URL"
+
+# github-deploy cleanup \
```
Comment (nmattia) [inline travis/deploy]: ?

File: swarm/deploy (hunk 1 above)
Comment (zimbatm) [reply swarm/deploy]: Everything is based on the PR number. Should I name it PR_ID instead?

File: travis/deploy (hunk 3 above)
Comment (zimbatm) [reply travis/deploy]: I'll activate this later, it still needs a bit of testing

File: swarm/lib.sh (hunk 2 above)
Comment (zimbatm) [reply swarm/lib.sh]: yeah, it's not strictly required either

**Comment by zimbatm** [comment]
> Sweet, could use some extra doc in the scripts though

I don't have a clear view of what is missing. Is it that it's hard to follow the deployment process? I also don't know the best place to put it so that you would read it.

**Comment by zimbatm** [comment]
I'm going to merge this for now and make further PRs to address your comments
